### PR TITLE
Add support for tar pax POSIX.1-2001 WebDataset

### PR DESCRIPTION
**Pull request by JanuszL**
- adds support for tar in pax POSIX.1-2001 in WebDataset

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Category:

**New feature** (*non-breaking change which adds functionality*)

## Description:
- adds support for tar in pax POSIX.1-2001 in WebDataset

### Affected modules and functionalities:
- webdataset loader
- webdataset reader tests

### Key points relevant for the review:
- NA

### Tests
- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [x] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: RDWDS.25

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/test/python/test_operator_readers_webdataset_requirements.py [hunk 1]
```diff
@@ -246,6 +246,46 @@ def test_sharding():
...
+    global test_batch_size
+    num_samples = 1000
+    tar_file_path = os.path.join(get_dali_extra_path(), "db/webdataset/MNIST/devel-0.tar")
+    pax_tar_file_path = os.path.join(get_dali_extra_path(), "db/webdataset/pax/devel-0.tar")
+    index_file = generate_temp_index_file(tar_file_path)
+
+    extract_dir = generate_temp_extract(tar_file_path)
+    equivalent_files = sorted(
+        glob(extract_dir.name + "/*"), key=lambda s: int(s[s.rfind("/") + 1 : s.rfind(".")])
+    )
```
Comment (klecki) [inline dali/test/python/test_operator_readers_webdataset_requirements.py:260]: This operation is not needed in this tests.

**Review by klecki — APPROVED** [review]
Unused code in test, otherwise looks ok.

File: dali/test/python/test_operator_readers_webdataset_requirements.py (hunk 1 above)
Comment (JanuszL) [reply dali/test/python/test_operator_readers_webdataset_requirements.py]: Fixed

**Review by klecki — APPROVED** [review]

**Comment by JanuszL** [comment]
!build

**Review by prak-nv — APPROVED** [review]

**Pull request merged**